### ISSUE TITLE
[Bug] Invalid hex code format does not revert back or prompt for an error 

### ISSUE THREAD

**Issue by ramsaptami** [appsmithorg/appsmith#6687]
## Description
When user inputs an invalid hex code, e.g. **#789945###** then colour does not change since it's an invalid code (_expected behaviour_) but it does not prompt for an error nor revert back to a valid hex code format (_expected behaviour that needs to be added_)

### Important Details
- Version: Cloud
- OS: Win10
- Browser: Chrome
- Environment: Production

https://www.loom.com/share/273a172c91334aa699a3fe3721673b7f

**Comment by AsianCat54x** [comment]
Hey, Can I work on this issue?

**Comment by vuiets** [comment]
## Stats
| Stat | Values |
| ------------- | ------------- |
| Reach     | 344 |
| Effort (months)  | 0.25 |

For reach,
People using the color property on a widget in the last 6 months: 1719
Expect about 20% to key in the wrong hex code value: 344

**Comment by rohitagarwal88** [comment]
@rahulramesha : Check the google chrome colour picker. We can follow the same approach here.
https://www.loom.com/share/a362767ba80043dbb0dbfa3719a92c06

**Comment by rahulramesha** [comment]
@rohitagarwal88 Throwing an error if it is the text does not resolve to a color might break people's apps, Something like the one you showed can work, like "not updating the color if it is an invalid color"

**Comment by vuiets** [comment]
Thinking out aloud. Would showing an error message for invalid color inputs in the debugger make sense here?

**Comment by rahulramesha** [comment]
@vuiets yes it does, But with the past experience on propertyPane controls, whenever a property was left unrestricted and then we decided to add restrictions it usually breaks user's apps. Have seen this happen too many times, wherein we had to roll back the restrictions.
@riodeuno what do you think?

**Comment by vuiets** [comment]
Agree with your observation here @rahulramesha. We don't have to do anything that is blocking or breaking the app here. The main problem here is nothing happens and one doesn't know what is wrong. So in case of a wrong input, we should atleast inform people that their entry is invalid.

Like you suggest we could not update the color when it is invalid but atleast warn them that it is invalid in the debugger so they are aware? Any downsides to doing that?

**Comment by rohitagarwal88** [comment]
[d2400b67-fba8-4322-b20e-621f62ec97a9.webm](https://user-images.githubusercontent.com/890915/176167882-f4cf5215-4d1f-4831-89e2-2bfba66d6d81.webm)
@vuiets / @rahulramesha : I agree with @vuiets opinions.
Check the retool error handling for color picker. That is inline with @vuiets expectation
